compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -526,11 +526,11 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
       case Function(args, body) =>
         this(this(x, args), body)
       case InfixOp(left, op, right) =>
-        this(this(x, left), right)
+        this(this(this(x, left), op), right)
       case PostfixOp(od, op) =>
-        this(x, od)
+        this(this(x, od), op)
       case PrefixOp(op, od) =>
-        this(x, od)
+        this(this(x, op), od)
```

Comment:
Does Dotty have prefix type operators? If so, how are they used?
"Regular" prefix operators already worked in the REPL *before* this change: it's a different ast that doesn't end up using the code above.

This is the hypothetical test I'd add, but like I said it's not really exercising this PR.

> scala> class B(val n: Int) {
>         def unary_! = new B(-n)
>        } 
> defined class B
> scala> val x = new B(42) 
> val x: B = B@10e92f8f
> scala> (!x).n 
> val res4: Int = -42

Thoughts?